x-pack/plugins/security_solution/server/lib/sourcerer/routes/index.ts

```diff
@@ -26,85 +35,220 @@ export const createSourcererDataViewRoute = (
+            ),
+          },
+        });
+      } catch (err) {
+        const error = transformError(err);
+        return siemResponse.error({
+          body:
+            error.statusCode === 403
+              ? 'Users with write permissions need to access the Elastic Security app to initialize the app source data.'
+              : error.message,
+          statusCode: error.statusCode,
+        });
+      }
+    }
+  );
+};
 
-        const activePatternLists = patternLists.map((pl, i) =>
-          // also remove duplicates from active
-          pl.filter((pattern, j, self) => self.indexOf(pattern) === j && activePatternBools[i][j])
+export const getSourcererDataViewRoute = (
+  router: SecuritySolutionPluginRouter,
+  getStartServices: StartServicesAccessor<StartPlugins>,
+  security: SetupPlugins['security']
+) => {
+  router.get(
+    {
+      path: SOURCERER_API_URL,
+      validate: {
+        query: buildRouteValidation(sourcererDataViewSchema),
+      },
+      options: {
+        authRequired: true,
+        tags: ['access:securitySolution'],
+      },
+    },
+    async (context, request, response) => {
+      const siemResponse = buildSiemResponse(response);
+      const siemClient = context.securitySolution?.getAppClient();
+      const { dataViewId } = request.query;
+      const dataViewBySecuritySolutionId = siemClient.getSourcererDataViewId();
+      const unsecuredSavedObjectClient = context.core.savedObjects.getClient({
+        excludedWrappers: ['security'],
+      });
+      try {
+        const [
+          ,
+          {
+            data: { indexPatterns },
+          },
+        ] = await getStartServices();
+        const auditLogger = security?.audit.asSystem(request);
+        /*
+         * Note for future engineer
+         * We need to have two different DataViewService because one will be to access all
+         * the data views that the user can access and the UnsecuredDataViewService will be
+         * to get the security solution data view
+         */
+        const unsecuredDataViewService = await indexPatterns.dataViewsServiceFactory(
+          unsecuredSavedObjectClient,
+          context.core.elasticsearch.client.asCurrentUser,
+          request,
+          true
+        );
+        const dataViewService = await indexPatterns.dataViewsServiceFactory(
+          context.core.savedObjects.client,
+          context.core.elasticsearch.client.asCurrentUser,
+          request
         );
 
-        const kibanaDataViews = allDataViews.map((kip, i) => ({
-          ...kip,
-          patternList: activePatternLists[i],
-        }));
-        const body = {
-          defaultDataView: kibanaDataViews.find((p) => p.id === dataViewId) ?? {},
-          kibanaDataViews,
-        };
-        return response.ok({ body });
+        let siemDataView;
+        if (dataViewId === dataViewBySecuritySolutionId) {
+          siemDataView = await unsecuredDataViewService.get(dataViewId);
```

Comment:
This feels wrong to me. The intent was to allow the data view to be _created_ without end-user privileges. Why are we allowing users to retrieve this without authorization?

Reply:
why, it is because we want to make sure the security solution work even if you do not have the privileges on the index patterns